test/passing/tests/module.ml

```diff
@@ -71,3 +71,12 @@ let helper ?x =
   match x with Some (module X : X_typ) -> X.f | None -> X_add_one.f
 
 let helper ?x:((module X) = (module X_add_one : X_typ)) = X.f
+
+module GZ : functor (X : sig end) () (Z : sig end) -> sig end =
+  (val Mooooooooooooooooooo)
+
+module GZ : functor (X : sig end) () (Z : sig end) -> sig end =
+  [%foooooooooooooooo]
+
+module GZZZZZZZZZZZZZZ : functor (X : sig end) () (Z : sig end) -> sig end =
+  _
```

Comment:
I think we definitely want this patch for `Pmod_hole` and `Pmod_unpack`.

Reply:
Alright I will only keep these 2 cases then, and we will wait until someone complains about the other 2 cases to see what we do.